### PR TITLE
TYT-17

### PR DESCRIPTION
**Pull request by JakMal9**
Growing and limit of planting to empty lands.

### CODE REVIEW

File: weltyt.py [hunk 1]
```diff
@@ -139,34 +139,39 @@ def buy_lands(lands):
         "growth_rate": 5,
         "price_ISL": 500,
         "plants": 0,
+        "growth_percentage": 0,
```
Comment (maciejsobolewski) [inline weltyt.py:142]: Is it needed here at all? (the same for `plants`)

File: weltyt.py [hunk 2]
```diff
@@ -139,34 +139,39 @@ def buy_lands(lands):
...
     },
     "clay_gravel": {
         "class": "clay_gravel",
         "id": 5,
         "growth_rate": 1,
         "price_ISL": 200,
         "plants": 0,
+        "growth_percentage": 0,
```
Comment (maciejsobolewski) [inline weltyt.py:174]: ditto x4

File: weltyt.py [hunk 3]
```diff
@@ -281,36 +287,48 @@ def buy_lands(lands):
         else:
 
             try:
```
Comment (maciejsobolewski) [inline weltyt.py:289]: We can try to move all this logic into separate function.

File: weltyt.py [hunk 4]
```diff
@@ -281,36 +287,48 @@ def buy_lands(lands):
...
-
-                    cur.execute(
-                        f'UPDATE lands SET plants = {new_plants_in_lands}  WHERE id = "{second_choice}";'
+                    planting_amount = input(
+                        "How many plants do you want to plant in your land?       "
                     )
+                    planting_amount = int(planting_amount)
+
+                    plants_in_lands = cur.execute(
+                        f'SELECT plants  FROM lands WHERE id = "{second_choice}";'
```
Comment (maciejsobolewski) [inline weltyt.py:306]: DRY.
Maybe we can modify `get_resource_value` function to handle such cases.

File: weltyt.py [hunk 5]
```diff
@@ -281,36 +287,48 @@ def buy_lands(lands):
...
+                        print("On one land you can plant only 100 plants")
+
+                    else:
 
-                    plants_in_resources = get_resource_value("amount", "Plants")
-                    new_plants_in_resources = plants_in_resources - planting_amount
-                    if new_plants_in_resources < 0:
                         cur.execute(
-                            f'UPDATE lands SET plants = {plants_in_lands}  WHERE id = "{second_choice}";'
+                            f'UPDATE lands SET plants = {new_plants_in_lands}  WHERE id = "{second_choice}";'
```
Comment (maciejsobolewski) [inline weltyt.py:316]: ditto

File: weltyt.py [hunk 6]
```diff
@@ -281,36 +287,48 @@ def buy_lands(lands):
...
-                        print("There is not enough plants to finish this operation.")
-                    else:
-                        update_resource_amount(new_plants_in_resources, "Plants")
 
-                        print(f"Well done planting in id {second_choice} land. ")
+                        plants_in_resources = get_resource_value("amount", "Plants")
+                        new_plants_in_resources = plants_in_resources - planting_amount
+                        if new_plants_in_resources < 0:
+                            cur.execute(
+                                f'UPDATE lands SET plants = {plants_in_lands}  WHERE id = "{second_choice}";'
```
Comment (maciejsobolewski) [inline weltyt.py:323]: Why are we updating anything here?

File: weltyt.py [hunk 7]
```diff
@@ -322,6 +340,30 @@ def buy_lands(lands):
 
     print("Go forward")
 
+    planted_lands = []
+    for p in cur.execute("SELECT * FROM lands WHERE plants>0"):
```
Comment (maciejsobolewski) [inline weltyt.py:344]: You can use list comprehension for that.

File: weltyt.py [hunk 8]
```diff
@@ -322,6 +340,30 @@ def buy_lands(lands):
...
+        planted_lands.append(p)
+
+    for g in planted_lands:
+        growing = g[5] + g[2] + 4
+        cur.execute(
+            f'UPDATE lands SET growth_percentage = {growing}  WHERE id = "{g[1]}";'
+        )
+
+    show_my_resources()
```
Comment (maciejsobolewski) [inline weltyt.py:353]: Debugging leftover?

File: weltyt.py [hunk 9]
```diff
@@ -322,6 +340,30 @@ def buy_lands(lands):
...
+
+    harvest = []
+    for grown in cur.execute("SELECT * FROM lands WHERE growth_percentage>100"):
+        harvest.append(grown)
```
Comment (maciejsobolewski) [inline weltyt.py:357]: ditto, list comprehension.

File: weltyt.py [hunk 10]
```diff
@@ -322,6 +340,30 @@ def buy_lands(lands):
...
+
+    for h in harvest:
+        cur.execute(
+            f'UPDATE resources SET amount = (SELECT amount FROM resources WHERE type = "Tobbaco_good") + (SELECT plants FROM lands WHERE id = "{h[1]}") WHERE type = "Tobbaco_good"'
```
Comment (maciejsobolewski) [inline weltyt.py:361]: I don't like that `Tobbaco_good` begin with capital letter :P

**Review by maciejsobolewski — CHANGES_REQUESTED** [review]

File: weltyt.py (hunk 7 above)
Comment (SMaciej) [reply weltyt.py:346]: `planted_lands = [p for p in cur.execute("SELECT * FROM lands WHERE plants>0")]` doesn't work?

File: weltyt.py (hunk 1 above)
Comment (SMaciej) [reply weltyt.py:150]: Yeah, but while buying we update db only, not this dictionary.
In fact, this dictionary never changes, am I right?

File: weltyt.py (hunk 7 above)
Comment (JakMal9) [reply weltyt.py:346]: Unfortunately, then planted_lands.append(p) doesn't work, and NameError occurs.

Comment (maciejsobolewski) [reply weltyt.py:346]: Why do you want to append anything? The point is you don't have to - list comprehension creates the list for you.

**Review by maciejsobolewski — APPROVED** [review]
Looks ok now, feel free to merge :)

**Pull request merged**